File: aoj/volume100/cpp/A10026.cpp
#include <cmath>
#include <cstdio>
#include <iostream>
#include <vector>

auto main() -> int {
	int n_scores;
	for (;;) {
		std::cin >> n_scores;
		if (n_scores == 0) {
			break;
		}

		std::vector<int> scores(n_scores);
		for (int i = 0; i < n_scores; ++i) {
			std::cin >> scores[i];
		}

		double average = 0.0;
		for (int score : scores) {
			average += score;
		}
		average /= n_scores;

		double variance = 0.0;
		for (int score : scores) {
			variance += std::pow(score - average, 2);
		}
		variance /= n_scores;

		double standard_deviation = std::sqrt(variance);
		std::printf("%.6f\n", standard_deviation);
	}
	return 0;
}

File: aoj/volume100/scala/A10026.scala
object Main {
  def main(args: Array[String]) {
    loop
  }

  def loop {
    val n = scala.io.StdIn.readLine
    if (n == "0") {
      return
    }

    val scores = (scala.io.StdIn.readLine split " ").map(_.toDouble)
    val average = scores.sum / scores.length
    val variance = scores.map(score => scala.math.pow(average - score, 2)).sum / scores.length
    val standard_deviation = scala.math.sqrt(variance)
    printf("%.6f\n", standard_deviation)
    loop
  }
}
